cmd/serve/main.go

```diff
@@ -32,8 +33,15 @@ func main() {
 	printRules(router.Rules)
 
 	port := config.ServerPort
+	server := &http.Server{
+		Addr:              ":" + port,
+		ReadHeaderTimeout: 3 * time.Second,
```

Comment:
I think we are still using the DefaultServeMux. We should not use it and pass in our own https://pkg.go.dev/net/http#Server

Its important because DefaultServeMux can be changed by any 3rd party lib we rely on.

Reply:
As you're aware from out slack chat, I'm in favor of making this change. I did play around with it but had some issues with configuring timeouts. This PR isn't introducing the fact that we're using the DefaultServeMux so let's get this merged and deal with that in another PR.

I created [this](https://dhis2.atlassian.net/browse/DEVOPS-222) jira issue so it (hopefully) won't just end with good intentions.